aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -69,6 +69,11 @@ at::Tensor& random_impl(at::Tensor& self, c10::optional<Generator> generator) {
 #define CHECK_OUT_OF_BOUNDS(var, name, min, max, dtype) \
   TORCH_CHECK(var >= min && var <= max, name , " is out of bounds for ", dtype); \
 
+#define CHECK_OUT_OF_BOUNDS_AND_SHOW_WARNING(var, name, min, max, dtype_min, dtype_max) \
+  if (var < min || var > max) { \
+    TORCH_WARN(name , " is out of bounds [", dtype_min, ", ", dtype_max, "]. This warning will become an error in version 1.7 release, please fix the code in advance"); \
```

Comment:
I think it would be better if you explained to users why this was out of bounds, because your new ranges are technically within the representable range for floating points, and the user will have to know something about floating point numbers to know what went wrong.

Reply:
And maybe a more consistent name for this macro would be `WARN_OUT_OF_BOUNDS`

Reply:
addressed